packages/relay/src/TransactionManager.ts

```diff
@@ -385,11 +393,15 @@ data                     | ${transaction.data}
     // Calculate new gas price as a % increase over the previous one, with a minimum value
     const gasFees = await this.contractInteractor.getGasFees()
     const { newMaxFee, newMaxPriorityFee, isMaxGasPriceReached } = this._resolveNewGasPrice(oldestPendingTx.maxFeePerGas, oldestPendingTx.maxPriorityFeePerGas, minMaxPriorityFee, parseInt(gasFees.baseFeePerGas))
-    const underpricedTransactions = sortedTxs.filter(it => it.maxFeePerGas < newMaxFee || it.maxPriorityFeePerGas < newMaxPriorityFee)
-    for (const transaction of underpricedTransactions) {
-      const boostedTransactionDetails = await this.resendTransaction(transaction, currentBlockHeight, newMaxFee, newMaxPriorityFee, isMaxGasPriceReached)
-      boostedTransactions.set(transaction.txId, boostedTransactionDetails)
-      this.logger.debug(`Replaced transaction: nonce: ${transaction.nonce} sender: ${signer} | ${transaction.txId} => ${boostedTransactionDetails.transactionHash}`)
+    for (const transaction of sortedTxs) {
+      // The tx is underpriced, boost it
+      if (transaction.maxFeePerGas < newMaxFee || transaction.maxPriorityFeePerGas < newMaxPriorityFee) {
+        const boostedTransactionDetails = await this.resendTransaction(transaction, currentBlockHeight, newMaxFee, newMaxPriorityFee, isMaxGasPriceReached)
+        boostedTransactions.set(transaction.txId, boostedTransactionDetails)
+        this.logger.debug(`Replaced transaction: nonce: ${transaction.nonce} sender: ${signer} | ${transaction.txId} => ${boostedTransactionDetails.transactionHash}`)
+      } else { // The tx is ok, just rebroadcast it
+        await this.resendTransaction(transaction, currentBlockHeight, transaction.maxFeePerGas, transaction.maxPriorityFeePerGas, transaction.maxFeePerGas > parseInt(this.config.maxGasPrice))
```

Comment:
do you count on the return value for anything? you didn't add resent TX there  (for all practical reasons, it was "boosted" )

Reply:
Boosted in this context also means txId change, so there's no need to return it I believe, since it's not a new transaction per se, but just a rebroadcast of the same one we have in storage.

Reply:
depends.
afaik, you don't use the returned txs for anything, except in test, where it is used to validate the relayer method did something - and return nothing means the method did nothing.
if this is the case, then re-broadcasting is still a "something" to report.
if tests don't use these return values anymore (e.g. test for side-effect using spies), then the ret values should be removed